src/middlewared/middlewared/plugins/smb.py

```diff
@@ -441,8 +441,11 @@ async def configure(self, job):
         into the registry.
         """
         job.set_progress(60, 'generating SMB share configuration.')
+        await self.middleware.call('cache.put', 'SMB_REG_INITIALIZED', False)
         await self.middleware.call("etc.generate", "smb_share")
         await self.middleware.call("smb.import_conf_to_registry")
+        await self.middleware.call('cache.put', 'SMB_REG_INITIALIZED', True)
+        os.unlink(SMBPath.SHARECONF.platform())
```

Comment:
Just wondered, is this always guaranteed to exist?

Reply:
We generate it in when we generate smb_share etc above. Reason for unlinking is to prevent user confusion regarding manual edits to this temporary file. We generate smb_share to basically prime the registry on SMB setup.